### PR TITLE
vdk-impala: Update README.md for vdk-impala

### PR DESCRIPTION
**Pull request by sbuldeev**
Why:
Enhance vdk-impala README.md

More details explained in #1361

What:
-Describe what quality checks are
-Provide example use

Signed-off-by: Stefan Buldeev sbuldeev@vmware.com

### CODE REVIEW

File: projects/vdk-plugins/vdk-impala/README.md [hunk 1]
```diff
@@ -26,6 +26,28 @@ For example
         job_input.execute_query("select 'Hi Impala!'")
 ```
 
+Most of the processing templates support quality checks that are used for preventing bad data going into production tables.
+The checks represent a callback function that is passed as an optional parameter to the job_input.execute_template() method.
```
Comment (DeltaMichael) [inline projects/vdk-plugins/vdk-impala/README.md:30]: I don't get this. How come some templates support this and others don't? It seems pretty generic to me, so shouldn't all templates support it?

Comment (sbuldeev) [reply projects/vdk-plugins/vdk-impala/README.md:30]: Currently quality checks handling is implemented only for scd1, snapshot and insert templates so users won't be able to use them for scd2 and versioned templates until they are implemented for them as well.

File: projects/vdk-plugins/vdk-impala/README.md [hunk 2]
```diff
@@ -26,6 +26,28 @@ For example
         job_input.execute_query("select 'Hi Impala!'")
 ```
 
+Most of the processing templates support quality checks that are used for preventing bad data going into production tables.
```
[the remark this answers is not in the document]
Comment (sbuldeev) [reply projects/vdk-plugins/vdk-impala/README.md:29]: Currently only scd1, snapshot and insert templates support quality checks. A user can find out if a template can be used with quality checks with checking the documentation on template level and see if there are check and staging table described in the parameters. I can add a few word on how to find out if a template supports quality checks?

[the remark this answers is not in the document]
Comment (sbuldeev) [reply projects/vdk-plugins/vdk-impala/README.md:29]: Sure, done that.

**Review by tozka — APPROVED** [review]

**Review by gageorgiev — APPROVED** [review]

**Review by DeltaMichael — APPROVED** [review]

**Pull request merged**